src/sas/qtgui/Plotting/QRangeSlider.py

```diff
@@ -0,0 +1,236 @@
+        """
+        self.clear_markers()
+        self.line_max.remove()
+        self.line_min.remove()
+
+    def update(self, x=None, y=None):
+        """
+        Draw the new roughness on the graph.
+        """
+        self.line_min.update(x, y)
+        self.line_max.update(x, y)
+        self.base.update()
+
+    def save(self, ev):
+        """
+        Remember the roughness for this layer and the next so that we
+        can restore on Esc.
+        """
+        self.line_min.save(ev)
+        self.line_max.save(ev)
+
+    def restore(self, ev):
+        """
+        Restore the roughness for this layer.
+        """
+        self.line_max.restore(ev)
+        self.line_min.restore(ev)
+
+    def move(self, x, y, ev):
+        """
+        Process move to a new position, making sure that the move is allowed.
+        """
+        pass
+
+    def clear_markers(self):
+        """
+        Should be no way to clear the markers
+        """
+        pass
+
+    def draw(self):
+        """
+        """
+        self.base.draw()
+
+
+class LineInteractor(BaseInteractor):
+    """
+    Draw a single vertical line that can be dragged on a plot
+    """
+    def __init__(self, base, axes, color='black', zorder=5, x=0.5, y=0.5, input=None, setter=None, getter=None):
+        """
+        """
+        BaseInteractor.__init__(self, base, axes, color=color)
+        self.base = base
+        self.markers = []
+        self.axes = axes
+        self.x = x
+        self.save_x = self.x
+        self.y_marker = y
+        self.save_y = self.y_marker
+        # Inner circle marker
+        self.inner_marker = self.axes.plot([self.x], [self.y_marker], linestyle='', marker='o', markersize=4,
+                                           color=self.color, alpha=0.6, pickradius=5, label=None, zorder=zorder,
+                                           visible=True)[0]
+        self.line = self.axes.axvline(self.x, linestyle='-', color=self.color, marker='', pickradius=5,
+                                      label=None, zorder=zorder, visible=True)
+        self.has_move = True
+        # Map GUI input to x value so slider and input update each other
+        self.input = input
+        if self.input:
+            self.input.textChanged.connect(self.inputChanged)
+        self.setter = setter if callable(setter) else None
+        self.getter = getter if callable(getter) else None
+        self.connect_markers([self.line, self.inner_marker])
+        self.update()
+
+    def validate(self, param_name, param_value):
+        """
+        Validate input from user - Should never fail
+        """
+        return True
+
+    def clear(self):
+        self.remove()
+
+    def remove(self):
+        """
+        Clear this slicer and its markers
+        """
+        self.inner_marker.remove()
+        self.line.remove()
+
+    def set_q(self, value):
+        """
+        Call the q setter callback method if it exists
+        """
+        if self.setter:
```

Comment:
You are actively testing for `self.setter` before each call to `set_q`. Is this extra check necessary then?

Reply:
I wanted redundancy to future proof this in case someone tries to call `set_q` or `get_q` from outside the class. I'll change them to private methods here and remove the check.